rfc/plugins.md

```diff
@@ -0,0 +1,300 @@
+
+The Front50 plugin registry will also be exposed through Gate, and can be consumed by Deck to load the correct plugins on application startup.
+
+Plugins can also be enabled or disabled at runtime without redeploying a service. This flag is sourced from Front50, which all services will be listening for change events from. Whether this change event is published or polled for initially is unknown.
+
+Halyard configs can be used to inform services of their _initial state_. When a service starts up, it will notify Front50 of what in-process plugins are installed, however Front50 will be responsible for determining the enabled state.
+
+Using Front50 as a source of truth will require extension of PF4J.
+
+```proto
+syntax = "proto3";
+
+// PluginManifest is used during registration of a plugin
+message PluginManifest {
+	// The name of the plugin, which is used as a fully 
+	// qualified identifier, including a namespace and 
+	// plugin name, ex: "netflix/fast-properties"
+	string name = 1;
+
+	// A short description of the plugin
+	// ex: "Provides rollout safety of runtime application 
+	// config via pipelines"
+	string description = 3;
+
+	// The name of the plugin author, ex: "Netflix"
+	string provider_name = 4;
+
+	// The semver plugin version, ex: "1.0.0"
+	string version = 5;
+
+	// A list of plugin dependencies. Initially this will be
+	// informational only. ex: "netflix/somedep>=1.x"
+	repeated string dependencies = 6;
+
+	// required_versions is a repeated string of service version	// requirements. e.g. "clouddriver >= 5.33.0"
+	repeated string required_versions = 7;
+
+	// A list of extension points that the plugin implements.
+	// e.g. "pipelineStage"
+	repeated string provided_capabilities = 8;
+
+	// Whether or not the plugin is enabled. An unset value 
+	// should be considered as true.
+	boolean enabled = 9;
+}
+```
+
+#### Sending Plugin Configuration to Deck
+
+While Front50 will be the source of truth for which plugins are enabled, we can pass that data via settings.js today and transition to using Front50 when it is ready.
+Using settings.js as an intermediate step allows us to collect feedback from selected users in order to improve the user experience in the future.
+
+Replacing/Updating the templating format for settings.js is out of scope for this project, so we propose to pass that data via a string that Deck can parse to determine which Plugins are installed.
+
+### Protobuf as the Remote Contract
+As a departure from PF4J, plugin contracts will be defined as Protobuf messages (not services), so long as they are _remote capable_. If a plugin is in-process only, and there’s no expectation that it becomes available for remote plugins, it does not need to define its contract as protobuf.
```

Comment:
This was a bit confusing to me, given the emphasis on P4J earlier in the doc, as well as the plugin manifest definition above in protobuf format (which, if I understand correctly, has nothing to do with the remote contract definition you speak of here). In any case, I think if we actually had the protocol (request/response?) message definitions here that would make things clearer.

Reply:
The exact shape of the protocol is not yet known.